src/compressed_chunk.c

```diff
@@ -81,4 +103,11 @@ ChunkIter_t *Compressed_NewChunkIterator(Chunk_t *chunk) {
 
 ChunkResult Compressed_ChunkIteratorGetNext(ChunkIter_t *iter, Sample* sample) {
   return Compressed_ReadNext((Compressed_Iterator *)iter, &sample->timestamp, &sample->value);
+}
+
+void Compressed_FreeChunkIterator(ChunkIter_t *iter, bool freeChunk) {
```

Comment:
@ashtul im not sure why we need `freeChunk` in both cases, I don't mind leaving it.

Reply:
Now that I saw `FREE_TEMP_CHUNK`, we should remove `freeChunk` from the API since the option flag is the actual marker if we need to release the chunk in the end of the iterator.

Reply:
Not sure what you mean.